hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/MemStoreCompactor.java

```diff
@@ -40,24 +40,25 @@
 @InterfaceAudience.Private
 class MemStoreCompactor {
 
-  // Option for external guidance whether flattening is allowed
-  static final String MEMSTORE_COMPACTOR_FLATTENING = "hbase.hregion.compacting.memstore.flatten";
-  static final boolean MEMSTORE_COMPACTOR_FLATTENING_DEFAULT = true;
-
-  // Option for external setting of the compacted structure (SkipList, CellArray, etc.)
+  // The external setting of the compacting MemStore behaviour
+  // Compaction of the index without the data is the default
   static final String COMPACTING_MEMSTORE_TYPE_KEY = "hbase.hregion.compacting.memstore.type";
-  static final int COMPACTING_MEMSTORE_TYPE_DEFAULT = 2;  // COMPACT_TO_ARRAY_MAP as default
+  static final String COMPACTING_MEMSTORE_TYPE_DEFAULT = "index-compaction";
 
   // What percentage of the duplications is causing compaction?
   static final String COMPACTION_THRESHOLD_REMAIN_FRACTION
       = "hbase.hregion.compacting.memstore.comactPercent";
-  static final double COMPACTION_THRESHOLD_REMAIN_FRACTION_DEFAULT = 0.2;
+  static final double COMPACTION_THRESHOLD_REMAIN_FRACTION_DEFAULT = 0.3;
```

Comment:
The speculative scan is still in the code. It is left only for the option when data-compaction is requested. It still can be avoided using MEMSTORE_COMPACTOR_AVOID_SPECULATIVE_SCAN. So yes this constant still have a meaning. It can be restructured later, under the "true policy JIRA".